### ISSUE TITLE
installation of dev version failed on windows 10

### ISSUE THREAD

**Issue by cderv** [r-lib/later#95]
Hi, 

I can't install dev version of `later` currently on my windows 10 setup. 
I am on R 3.6.1 with Rtools version 3.5.0.4 

I tried 4 approaches to get different logs and locate issues and potential solutions;

## `remotes::install_github`

I get this messages
```r
> remotes::install_github("r-lib/later")
Downloading GitHub repo r-lib/later@master
r-lib-later-e379210/include: Can't create '\\\\?\\C:\\Users\\chris\\AppData\\Local\\Temp\\RtmpMDDLNd\\remotesb88891e1a4e\\r-lib-later-e379210\\include'
tar.exe: Error exit delayed from previous errors.
Skipping 2 packages not available: Rcpp, BH
√  checking for file 'C:\Users\chris\AppData\Local\Temp\RtmpMDDLNd\remotesb88891e1a4e\r-lib-later-e379210/DESCRIPTION' (678ms)
-  preparing 'later': (369ms)
√  checking DESCRIPTION meta-information ... 
-  cleaning src
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
-  building 'later_0.8.0.9002.tar.gz'
   
Installing package into 'C:/Users/chris/Documents/R/win-library/3.6'
(as 'lib' is unspecified)
Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
* installing *source* package 'later' ...
** using staged installation
** libs

*** arch - i386
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c RcppExports.cpp -o RcppExports.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c callback_registry.cpp -o callback_registry.o
In file included from C:/Rtools/mingw_32/i686-w64-mingw32/include/windows.h:71:0,
                 from tinycthread.h:104,
                 from threadutils.h:10,
                 from callback_registry.h:11,
                 from callback_registry.cpp:5:
debug.h:36:3: error: expected identifier before numeric constant
   ERROR,
   ^
debug.h:36:3: error: expected '}' before numeric constant
debug.h:36:3: error: expected unqualified-id before numeric constant
In file included from callback_registry.cpp:6:0:
debug.h:40:1: error: expected declaration before '}' token
 };
 ^
make: *** [C:/PROGRA~1/R/R-3.6.1/etc/i386/Makeconf:215: callback_registry.o] Error 1
ERROR: compilation failed for package 'later'
* removing 'C:/Users/chris/Documents/R/win-library/3.6/later'
* restoring previous 'C:/Users/chris/Documents/R/win-library/3.6/later'
Error: Failed to install 'later' from GitHub:
  (converted from warning) installation of package 'C:/Users/chris/AppData/Local/Temp/RtmpMDDLNd/fileb888246f362f/later_0.8.0.9002.tar.gz' had non-zero exit status
In addition: Warning message:
In utils::untar(tarfile, ...) :
  'tar.exe -xf "C:\Users\chris\AppData\Local\Temp\RtmpMDDLNd\fileb888684b62fb.tar.gz" -C "C:/Users/chris/AppData/Local/Temp/RtmpMDDLNd/remotesb88891e1a4e"' returned error code 1
Call `rlang::last_error()` to see a backtrace
```

## Cloning repo and `remotes::install_local`

```r
> remotes::install_local()
Skipping 2 packages not available: Rcpp, BH
√  checking for file 'C:\Users\chris\AppData\Local\Temp\RtmpMDDLNd\fileb88813ee76aa\later/DESCRIPTION' (384ms)
-  preparing 'later': (444ms)
√  checking DESCRIPTION meta-information ... 
-  cleaning src
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
-  building 'later_0.8.0.9002.tar.gz'
   
Installing package into 'C:/Users/chris/Documents/R/win-library/3.6'
(as 'lib' is unspecified)
Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
* installing *source* package 'later' ...
** using staged installation
** libs

*** arch - i386
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c RcppExports.cpp -o RcppExports.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c callback_registry.cpp -o callback_registry.o
In file included from C:/Rtools/mingw_32/i686-w64-mingw32/include/windows.h:71:0,
                 from tinycthread.h:104,
                 from threadutils.h:10,
                 from callback_registry.h:11,
                 from callback_registry.cpp:5:
debug.h:36:3: error: expected identifier before numeric constant
   ERROR,
   ^
debug.h:36:3: error: expected '}' before numeric constant
debug.h:36:3: error: expected unqualified-id before numeric constant
In file included from callback_registry.cpp:6:0:
debug.h:40:1: error: expected declaration before '}' token
 };
 ^
make: *** [C:/PROGRA~1/R/R-3.6.1/etc/i386/Makeconf:215: callback_registry.o] Error 1
ERROR: compilation failed for package 'later'
* removing 'C:/Users/chris/Documents/R/win-library/3.6/later'
* restoring previous 'C:/Users/chris/Documents/R/win-library/3.6/later'
Error: Failed to install 'later' from local:
  (converted from warning) installation of package 'C:/Users/chris/AppData/Local/Temp/RtmpMDDLNd/fileb88866f1357c/later_0.8.0.9002.tar.gz' had non-zero exit status
Call `rlang::last_error()` to see a backtrace
```

## Cloning and just `pkgbuild::build`

```r
> pkgbuild::build()
√  checking for file 'C:\Users\chris\Documents\DEV_R\later/DESCRIPTION' (387ms)
-  preparing 'later': (1s)
√  checking DESCRIPTION meta-information ... 
-  cleaning src
-  installing the package to build vignettes (346ms)
         -----------------------------------
-  installing *source* package 'later' ...
   ** using staged installation
   ** libs
   C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c RcppExports.cpp -o RcppExports.o
   C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c callback_registry.cpp -o callback_registry.o
   In file included from C:/Rtools/mingw_64/x86_64-w64-mingw32/include/windows.h:71:0,
                    from tinycthread.h:104,
                    from threadutils.h:10,
                    from callback_registry.h:11,
                    from callback_registry.cpp:5:
   debug.h:36:3: error: expected identifier before numeric constant
      ERROR,
      ^
   debug.h:36:3: error: expected '}' before numeric constant
   debug.h:36:3: error: expected unqualified-id before numeric constant
   In file included from callback_registry.cpp:6:0:
   debug.h:40:1: error: expected declaration before '}' token
    };
    ^
   make: *** [C:/PROGRA~1/R/R-3.6.1/etc/x64/Makeconf:215: callback_registry.o] Error 1
   ERROR: compilation failed for package 'later'
-  removing 'C:/Users/chris/AppData/Local/Temp/RtmpyqAj6c/Rinst17b820d9105e/later'
         -----------------------------------
   ERROR: package installation failed
Error in (function (command = NULL, args = character(), error_on_status = TRUE,  : 
  System command error
Call `rlang::last_error()` to see a backtrace
```

## Cloning and install from RStudio 

Running `Install & Restart` in RStudio gave me this error
```r
==> Rcpp::compileAttributes()

* Updated R/RcppExports.R

==> Rcmd.exe INSTALL --no-multiarch --with-keep.source later

Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
* installing to library 'C:/Users/chris/Documents/R/win-library/3.6'
* installing *source* package 'later' ...
** using staged installation
** libs
make: *** No rule to make target '$(CXX11STD)''.  Stop.
ERROR: compilation failed for package 'later'
* removing 'C:/Users/chris/Documents/R/win-library/3.6/later'
* restoring previous 'C:/Users/chris/Documents/R/win-library/3.6/later'

Exited with status 1.
```

The error form RStudio build pane is really different - I don't know why. maybe a config for RStudio to find Rtools.

I can't build to. Clicking on Build Source Package I get: 
```r
==> Rcpp::compileAttributes()

* Updated R/RcppExports.R

==> devtools::document(roclets = c('rd', 'collate', 'namespace'))

Updating later documentation
Loading later
Error in .Call("_later_ensureInitialized", PACKAGE = "later") : 
  "_later_ensureInitialized" non disponible pour .Call() pour le package"later"
Calls: suppressPackageStartupMessages ... run_pkg_hook -> <Anonymous> -> ensureInitialized -> .Call
De plus : Warning message:
roxygen2 requires Encoding: UTF-8 
Ex�cution arr�t�e

Exited with status 1.
```

## What to do ? 

I think there is something to do within Makevars and/or a configure  and/or some `.h` files so that it works for windows too. I am not enough expert in C++ compilation to try something without guidance.
Can you help on that or do you plan to deal with windows compilation later ?

**Comment by wch** [comment]
Can you try #96?

```R
devtools::install_github('r-lib/later#96')
```

**Comment by cderv** [comment]
That works fine now with `install_github`, Thanks !  

<details><summary>`devtools::install_github('r-lib/later#96')` log </summary>

```r
> devtools::install_github('r-lib/later#96')
Downloading GitHub repo r-lib/later@fix-log-prefix
r-lib-later-ed5d104/include: Can't create '\\\\?\\C:\\Users\\chris\\AppData\\Local\\Temp\\Rtmp8gLcuO\\remotes4ef473dd564c\\r-lib-later-ed5d104\\include'
tar.exe: Error exit delayed from previous errors.
Skipping 2 packages not available: Rcpp, BH
√  checking for file 'C:\Users\chris\AppData\Local\Temp\Rtmp8gLcuO\remotes4ef473dd564c\r-lib-later-ed5d104/DESCRIPTION' (373ms)
-  preparing 'later':
√  checking DESCRIPTION meta-information ... 
-  cleaning src
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
-  building 'later_0.8.0.9002.tar.gz'
   
Installing package into 'C:/Users/chris/Documents/R/win-library/3.6'
(as 'lib' is unspecified)
Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
* installing *source* package 'later' ...
** using staged installation
** libs

*** arch - i386
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c RcppExports.cpp -o RcppExports.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c callback_registry.cpp -o callback_registry.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c debug.cpp -o debug.o
debug.cpp: In function 'std::string log_level(std::string)':
debug.cpp:63:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
C:/Rtools/mingw_32/bin/gcc  -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O3 -Wall  -std=gnu99 -mtune=generic -c init.c -o init.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c later.cpp -o later.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c later_posix.cpp -o later_posix.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c later_win32.cpp -o later_win32.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c timer_posix.cpp -o timer_posix.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c timestamp_unix.cpp -o timestamp_unix.o
C:/Rtools/mingw_32/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c timestamp_win32.cpp -o timestamp_win32.o
C:/Rtools/mingw_32/bin/gcc  -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O3 -Wall  -std=gnu99 -mtune=generic -c tinycthread.c -o tinycthread.o
C:/Rtools/mingw_32/bin/g++ -shared -s -static-libgcc -o later.dll tmp.def RcppExports.o callback_registry.o debug.o init.o later.o later_posix.o later_win32.o timer_posix.o timestamp_unix.o timestamp_win32.o tinycthread.o -pthread -LC:/Users/chris/AppData/Local/renv/extsoft/lib/i386 -LC:/Users/chris/AppData/Local/renv/extsoft/lib -LC:/PROGRA~1/R/R-3.6.1/bin/i386 -lR
installing to C:/Users/chris/Documents/R/win-library/3.6/00LOCK-later/00new/later/libs/i386

*** arch - x64
C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c RcppExports.cpp -o RcppExports.o
C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c callback_registry.cpp -o callback_registry.o
C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c debug.cpp -o debug.o
debug.cpp: In function 'std::string log_level(std::string)':
debug.cpp:63:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
C:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=generic -c init.c -o init.o
C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c later.cpp -o later.o
C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c later_posix.cpp -o later_posix.o
C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c later_win32.cpp -o later_win32.o
C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c timer_posix.cpp -o timer_posix.o
C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c timestamp_unix.cpp -o timestamp_unix.o
C:/Rtools/mingw_64/bin/g++  -std=gnu++11 -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -mtune=generic -c timestamp_win32.cpp -o timestamp_win32.o
C:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-3.6.1/include" -DNDEBUG -pthread -DSTRICT_R_HEADERS -I"C:/Users/chris/Documents/R/win-library/3.6/Rcpp/include" -I"C:/Users/chris/Documents/R/win-library/3.6/BH/include"   -I"C:/Users/chris/AppData/Local/renv/extsoft/include"     -O2 -Wall  -std=gnu99 -mtune=generic -c tinycthread.c -o tinycthread.o
C:/Rtools/mingw_64/bin/g++ -shared -s -static-libgcc -o later.dll tmp.def RcppExports.o callback_registry.o debug.o init.o later.o later_posix.o later_win32.o timer_posix.o timestamp_unix.o timestamp_win32.o tinycthread.o -pthread -LC:/Users/chris/AppData/Local/renv/extsoft/lib/x64 -LC:/Users/chris/AppData/Local/renv/extsoft/lib -LC:/PROGRA~1/R/R-3.6.1/bin/x64 -lR
installing to C:/Users/chris/Documents/R/win-library/3.6/00LOCK-later/00new/later/libs/x64
** R
** inst
** byte-compile and prepare package for lazy loading
Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6/00LOCK-later/00new
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
** help
*** installing help indices
  converting help for package 'later'
    finding HTML links ... fini
    create_loop                             html  
    later                                   html  
    list_queue                              html  
    logLevel                                html  
    loop_empty                              html  
    next_op_secs                            html  
    run_now                                 html  
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
*** arch - i386
Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6/00LOCK-later/00new
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
*** arch - x64
Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6/00LOCK-later/00new
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
** testing if installed package can be loaded from final location
*** arch - i386
Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
*** arch - x64
Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
** testing if installed package keeps a record of temporary installation path
* DONE (later)
Warning message:
In utils::untar(tarfile, ...) :
  'tar.exe -xf "C:\Users\chris\AppData\Local\Temp\Rtmp8gLcuO\file4ef44c4245e2.tar.gz" -C "C:/Users/chris/AppData/Local/Temp/Rtmp8gLcuO/remotes4ef473dd564c"' returned error code 1
```

</details>

There is still an error while installing of building from RStudio IDE build pane. 

<details><summary> Install & Restart </summary>

```r
==> Rcpp::compileAttributes()

* Updated R/RcppExports.R

==> Rcmd.exe INSTALL --no-multiarch --with-keep.source later

Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
* installing to library 'C:/Users/chris/Documents/R/win-library/3.6'
* installing *source* package 'later' ...
** using staged installation
** libs
make: *** No rule to make target '$(CXX11STD)''.  Stop.
ERROR: compilation failed for package 'later'
* removing 'C:/Users/chris/Documents/R/win-library/3.6/later'
* restoring previous 'C:/Users/chris/Documents/R/win-library/3.6/later'

Exited with status 1.
```
</details>

<details><summary>Build Source </summary>

```r
==> Rcpp::compileAttributes()

* Updated R/RcppExports.R

==> devtools::document(roclets = c('rd', 'collate', 'namespace'))

Updating later documentation
Writing NAMESPACE
Loading later
Error in .Call("_later_ensureInitialized", PACKAGE = "later") : 
  "_later_ensureInitialized" non disponible pour .Call() pour le package"later"
Calls: suppressPackageStartupMessages ... run_pkg_hook -> <Anonymous> -> ensureInitialized -> .Call
De plus : Warning message:
roxygen2 requires Encoding: UTF-8 
Ex�cution arr�t�e

Exited with status 1.
```
</details>

For the later, `pkgbuild::build` is working, it is `devtools::document` which give the error. 

Should I open new issues for those ? 
It is not important for me right now as I can install chromote and pagedown PR now, but still a behavior you may want to fix. 🤔

**Comment by wch** [comment]
@cderv Hm, I'm not sure about that CXX11STD issue. Can you also try the `win-fixes` branch and let me know if it works?

Also, what version of Rtools are you using? (I believe they're still using rtools34 on CRAN.)

**Comment by cderv** [comment]
I am currently Rtools version 3.5.0.4 last one from CRAN. 
https://cran.r-project.org/bin/windows/Rtools/VERSION.txt

I can't try again just right now. I'll do that later today and tell you.

**Comment by cderv** [comment]
I tried the branch and it does not work.

Going back to the PR branch, I try several things

* Building a bundle then installing is working. (`devtools::install_local()`, `devtools::install_github`)
* Not building a bundle but installing from directory works (what does `devtools::install_local(build = FALSE)`. 
* Installing using R CMD INSTALL is what throws the CXX11STD error (used using `devtools::install`)
I guess install & Reload in RStudio IDE build pane is using this, hence the error. 
* using `install.packages(".", repos = NULL, type = "source")` throws the same error.

<details><summary> devtools::install </summary>

> devtools::install()
Skipping 2 packages not available: Rcpp, BH
√  checking for file 'C:\Users\chris\Documents\DEV_R\later/DESCRIPTION' (346ms)
-  preparing 'later': (1.1s)
√  checking DESCRIPTION meta-information ... 
-  cleaning src
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
-  building 'later_0.8.0.9002.tar.gz'
   
Running "C:/PROGRA~1/R/R-3.6.1/bin/x64/Rcmd.exe" INSTALL \
  "C:\Users\chris\AppData\Local\Temp\Rtmp00qj4q/later_0.8.0.9002.tar.gz" --install-tests 
* installing to library 'C:/Users/chris/Documents/R/win-library/3.6'
* installing *source* package 'later' ...
** using staged installation
** libs
-
*** arch - i386
make: *** No rule to make target '$(CXX11STD)''.  Stop.
ERROR: compilation failed for package 'later'
* removing 'C:/Users/chris/Documents/R/win-library/3.6/later'
* restoring previous 'C:/Users/chris/Documents/R/win-library/3.6/later'
```
</details>

<details><summary> install.packages(".", repos = NULL, type = "source")</summary>

> install.packages(".", repos = NULL, type = "source")
Installing package into 'C:/Users/chris/Documents/R/win-library/3.6'
(as 'lib' is unspecified)
Using libraries at paths:
- C:/Users/chris/Documents/R/win-library/3.6
- C:/Program Files/R/R-3.6.1/library
* installing *source* package 'later' ...
** using staged installation
** libs

*** arch - i386
make: *** No rule to make target '$(CXX11STD)''.  Stop.
ERROR: compilation failed for package 'later'
* removing 'C:/Users/chris/Documents/R/win-library/3.6/later'
* restoring previous 'C:/Users/chris/Documents/R/win-library/3.6/later'
Warning in install.packages :
  installation of package '.' had non-zero exit status
```
</details>

I think this all come from a  windows / R / RStudio configuration. 

devtools and remote uses `pkgbuild` to prepare the environment for package installation using Rtools. 

The conclusion is : 

* `install.packages(".", repos = NULL, type = "source")` throws the CXX11STD error
* `pkgbuild::with_build_tools(install.packages(".", repos = NULL, type = "source"))` works. 

I guess RStudio IDE does not prepare the same way as `pkgbuild`, I thought it would be the same.  
Strangely, I have Rtools in the PATH already, but not first in the PATH as pkgbuild does by putting Rtools path first. 

There must be some mysterious configuration stuff. I'll see if this happens with other package. 

I think you can close this issue as PR #96 fix the installation issue.

**Comment by wch** [comment]
Thanks for testing! I merged the PR but I'll keep this issue open to remind us to investigate further.

**Comment by wch** [comment]
For the record, I just tested the following with R 3.5.1 and Rtools 3.5.0.4 on Windows, and they worked fine:

* `devtools::install_github('r-lib/later')`
* `devtools::install()`
* RStudio's Build and Install command
* `install.packages(".", repos = NULL, type = "source")` also worked, but in general I would not expect this to work (since it expects either a source package or the unzipped contents of a source package, which is not exactly the same as the raw source).

**Comment by cderv** [comment]
So this has something to do with my setup and how Rtools, R and RStudio are configured together.
As I got the hint last time, It seems to be something with where is Rtools in the PATH. 

I just tried now with 3.5.1 like you and 3.6.1 as I did before by putting RTools **first** in the PATH this time and it worked. This is what pkgbuild does in fact and why it works when using it. 

Sorry for the delay on this one. One on two issues was correct... 😉 
And thanks a lot for trying.
